qcodes/dataset/data_set.py

```diff
@@ -753,6 +799,38 @@ def add_results(self, results: Sequence[Mapping[str, VALUE]]) -> int:
                            values)
         return len_before_add
 
+    def add_result_to_queue(self,
```

Comment:
What will happen is someone calls `add_result_to_query` for a dataset that is not created with `start_bg_writer`=True ?

Reply:
this comment made me wonder whether the "queue" part should be explicit on datasaver's and dataset's public interface (as these methods like `add_result_to_queue`) or should this be hidden behind the existing methods like `add_results`?

Reply:
The two bullet points in the comment below addressing the concerns here have now been implemented.